pkg/common/resource.go

```diff
@@ -87,6 +86,48 @@ func GetPodResource(pod *v1.Pod) (resource *si.Resource) {
 	return podResource
 }
 
+func containerResource(pod *v1.Pod, i int) (resource *si.Resource) {
+	// K8s pod InPlacePodVerticalScaling from:
+	// alpha: v1.27
+	// beta: v1.31?
+	// If AllocatedResources are present, these need to be used in preference to pod resource requests.
+	// Additionally, if the Resize pod status is Proposed, then the maximum of the request and allocated values need
+	// to be used.
+	requested := pod.Spec.Containers[i].Resources.Requests
+	if len(pod.Status.ContainerStatuses) == 0 {
+		return getResource(requested)
+	}
+	allocated := pod.Status.ContainerStatuses[i].AllocatedResources
+	if len(allocated) == 0 {
+		// no allocatedResources present, use requested
+		return getResource(requested)
+	}
+	if pod.Status.Resize == v1.PodResizeStatusProposed {
```

Comment:
from the spec: `For containers which have Status.Resize = "Proposed", it must be pessimistic and assume that the resize will be imminently accepted. Therefore it must use the larger of the Pod's Spec...Resources.Requests and Status...AllocatedResources values`

Should we replace `PodResizeStatusInProgress` by `PodResizeStatusProposed`?

Reply:
oh, you have updated the code. please ignore my comment.

Reply:
Yes, I spotted that while re-reading the spec.